pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -85,7 +85,8 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, r *http.Requ
 		switch doc.OpenShiftCluster.Properties.FailedProvisioningState {
 		case api.ProvisioningStateCreating:
 			return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeRequestNotAllowed, "", "Request is not allowed on cluster whose creation failed. Delete the cluster.")
-		case api.ProvisioningStateUpdating:
+		case api.ProvisioningStateUpdating, api.ProvisioningStateAdminUpdating:
+			// TODO: Discuss. Do we want users to be able to do operations on clusters in api.ProvisioningStateAdminUpdating failed state
```

Comment:
Note the comment

Reply:
Need to think about that, I agree.

Reply:
For now, I suggest that we disallow users from updating a cluster in Failed/AdminUpdating.  My rationale is:
a) that if the update fails, the cluster probably goes into Failed/Updating and we lose track that the admin update caused the first failure
b) we can open this up later if we need to